src/native/libs/System.Native/pal_networking.c

```diff
@@ -1186,6 +1186,28 @@ int32_t SystemNative_GetIPv4MulticastOption(intptr_t socket, int32_t multicastOp
     return Error_SUCCESS;
 }
 
+#ifdef __APPLE__
+static int32_t Apple_SetIPv4MulticastInterface(int fd, IPv4MulticastOption* option)
+{
+    if (option->InterfaceIndex != 0)
+    {
+        // On Apple platforms, IP_MULTICAST_IF with interface index uses the interface index directly
+        uint32_t ifindex = (uint32_t)option->InterfaceIndex;
+        int err = setsockopt(fd, IPPROTO_IP, IP_MULTICAST_IFINDEX, &ifindex, sizeof(ifindex));
```

Comment:
I think we should still set address too here. I don't think the interface alone is sufficient.

Reply:
I didn't understand what you mean, I tested it, it works as expected, setting address for MulticastInterface, just helps us to detect which interface we should choose if I understood that correctly.